core/src/main/java/org/apache/iceberg/ReachableFileCleanup.java

```diff
@@ -208,8 +231,16 @@ private Set<String> findFilesToDelete(
                 }
 
                 // Remove all the live files from the candidate deletion set
-                try (CloseableIterable<String> paths = ManifestFiles.readPaths(manifest, fileIO)) {
-                  paths.forEach(filesToDelete::remove);
+                try (ManifestReader<DataFile> reader =
+                    ManifestFiles.read(manifest, fileIO)
+                        .select(ImmutableList.of("content", "file_path"))) {
+                  reader
+                      .liveEntries()
+                      .forEach(
+                          entry -> {
+                            DataFile file = entry.file();
+                            filesToDelete.remove(new FileInfo(file.content(), file.location()));
+                          });
```

Comment:
can extract this in a common util as it same as  198-207

Reply:
its not exactly the same, but i have extracted some of the common part now